### PR TITLE
sdk fixes

### PR DESCRIPTION
**Pull request by ocke**
:v:

/cc @zendesk/vegemite

### Description
Fxies some array in array wrapping
https://github.com/zendesk/zendesk_app_framework/pull/574 needs to be deployed first.

### References
* JIRA: 

### Risks

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -14,10 +14,10 @@ function rawPostMessage(client, msg, forceReady) {
   }
 }
 
-function wrappedPostMessage(name) {
+/* params is an array for set/invoke or an object for set */
```
Comment (svizzari) [inline lib/client.js]: How come this one line is a block comment?

Comment (ocke) [inline lib/client.js]: because it starts with `/*` and ends with `*/`

Comment (danielbreves) [inline lib/client.js]: did you mean `get` instead of `set`?

File: lib/client.js [hunk 2]
```diff
@@ -14,10 +14,10 @@ function rawPostMessage(client, msg, forceReady) {
   }
 }
 
-function wrappedPostMessage(name) {
+// params is an array for get/invoke or an object for set
+function wrappedPostMessage(name, params) {
```
Comment (danielbreves) [inline lib/client.js]: I realise that this is an existing issue, but I don't like the fact the `params` array represents multiple calls for `get` but a single call for `invoke`.

File: lib/client.js [hunk 3]
```diff
@@ -303,22 +303,46 @@ Client.prototype = {
     }.bind(this));
   },
 
+  // Accepts string or array of strings.
   get: function(path) {
-    var paths = Array.isArray(path) ? path : [path];
+    var paths = Array.isArray(path) ? path : Array.prototype.slice.call(arguments, 0);
```
Comment (adammw) [inline lib/client.js]: i'm skeptical

File: lib/client.js [hunk 4]
```diff
@@ -303,22 +303,46 @@ Client.prototype = {
...
-  invoke: function() {
-    return wrappedPostMessage.bind(this, 'invoke').apply(null, arguments);
+  invoke: function(key) {
+    var obj = key;
+
+    if (typeof key === 'string') {
+      obj = {};
+      obj[key] = Array.prototype.slice.call(arguments, 1);
+    } else {
+      throw new Error('Invoke with an object isn\'t supported (yet).');
```
Comment (adammw) [inline lib/client.js]: let's not say yet, that implies that we will ever get around to fixing it

**Comment by adammw** [comment]
I'd prefer to leave out the multiple argument conversion into an array for the get, seems a bit too magic for our public SDK and limits us if we want to add other arguments in the future that do different things. It's shouldn't be that hard for developers to just create and use an array. 
That being said, I can be swayed if the rest of the team wants this.

**Comment by adammw** [comment]
:+1:

**Comment by danielbreves** [comment]
👍  but let's not merge this until zendesk/zendesk_app_framework#574 is in production.

**Comment by ocke** [comment]
but it will only create beta4?

**Comment by danielbreves** [comment]
ah right, I guess it's ok then

**Comment by ocke** [comment]
@danielbreves merge? framework is in production.

**Comment by adammw** [comment]
:+1:

**Pull request merged**